server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -187,14 +187,15 @@ public void parse(DocumentParserContext context) throws IOException {
                 } else if (token == Token.VALUE_NUMBER || token == Token.VALUE_STRING) {
                     final String key = name() + "." + feature;
                     float value = context.parser().floatValue(true);
-                    if (context.doc().getByKey(key) != null) {
-                        throw new IllegalArgumentException(
-                            "[sparse_vector] fields do not support indexing multiple values for the same feature ["
-                                + key
-                                + "] in the same document"
+
+                    // if we have an existing feature of the same name we'll select for the one with the max value
+                    // based on recommendations from this paper: https://arxiv.org/pdf/2305.18494.pdf
```

Comment:
👍

Reply:
So the best way to score long docs is to take the max over scores.

Suppose you have a document with passages `p_1`, `p_2`, ..., `p_n` and query `q` then you would score the document for the query as
```
score_best = max_i(q^t p_i) = max_i(sum_j(w_j(q)) x w_j(p_i)))
```
The second best way to score the document is to use max pooling of weights
```
score_runnerup = sum_j(w_j(q) x max_i(w_j(p_i)))
```
where `w_j` is the function returning the j'th token weight. I may be missing the context, but my understanding here is that feature refers to `w_j` so we use the second approach. This is ok, but it tends to suffer more on docs with many passages than the first approach.

All that said, if you are indexing the joint representation, i.e. single sparse vector with max token weights to provide a tradeoff for index size and quality then this is absolutely fine. My comment only applies if this is the only way we are considering scoring long docs from their passages.

Reply:
@tveasey 

> My comment only applies if this is the only way we are considering scoring long docs from their passages.

Definitely not. Folks can still use `nested` if they so wish.